src/App.tsx

```diff
@@ -8,55 +8,163 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { customFetch } from './utils/fetchClient';
+import { User } from './types/User';
+import { ErrorType } from './types/ErrorType';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isErrorType, setErrorType] = useState<ErrorType | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isPostListVisible, setIsPostListVisible] = useState(false);
+  const [isLoaderVisible, setIsLoaderVisible] = useState(false);
+  const [isCommentFetch, setIsCommentFetch] = useState(false);
+  const [isCommentListHidden, setIsCommentListHidden] = useState(true);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isOpenFormForComment, setOpenFormForComment] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const loadingErrors = [ErrorType.USERS, ErrorType.POSTS];
+
+  const getUsers = async () => {
+    try {
+      const fetchUsers = await customFetch.getUserFromServer();
+
+      setUsers(fetchUsers as User[]);
+      setErrorType(null);
+    } catch (error) {
+      setErrorType(ErrorType.USERS);
+    }
+  };
+
+  useEffect(() => {
+    getUsers();
+  }, []);
+
+  const getUserPostsFromServer = async (userId: number) => {
+    setIsLoaderVisible(true);
+    setIsPostListVisible(false);
+
+    try {
+      const fetchPost = await customFetch.getPostFromServer(userId);
+
+      setErrorType(null);
+
+      setPosts(fetchPost);
+      setIsPostListVisible(true);
+      setErrorType(null);
+    } catch (error) {
+      setErrorType(ErrorType.POSTS);
+      setIsPostListVisible(false);
+    } finally {
+      setIsLoaderVisible(false);
+    }
+  };
+
+  const getCommentByPostFromServer = async (postId: number) => {
```

Comment:
Consider implementing error handling for the case when the fetchComments request fails silently. Check if fetchComments is an array before setting the state.